### DISCUSSION TITLE
Global persistent variables in NextJS

### DISCUSSION THREAD

**Discussion by mdjfs** [vercel/next.js#59543] [Help]
### Summary

Sometimes, i need some way to store persistent javascript objects to use in RSC/Api endpoints. For example, saving an instance of a object that takes me a long time to build, and be more efficient when i need use that object for any process. Actually we have the "global" object. But is cleaned randomly. Isn't really persistent. Could be very useful if we have some way to have persistent javascript objects to use in any server side part of the application

**Comment by iki** [comment]
I only found `process.env` to be persistent. Even `globalThis` is not

**Reply by mdjfs** [reply]
It could work, but I don't know if assigning a variable to the `process.env` could eventually be cleaned

**Reply by iki** [reply]
@mdjfs in my tests, `process.env` was the only global variable that persisted values across multiple requests. It only reset on Next.js dev server reload after detecting source code changes.

However, it is limited to string keys and values (I used JSON to store objects), and it also has limits by OS like total size (32k on Windows), max key length (255 on all Win/Mac/Linux afaik) and allowed key characters (recommended `A-Z, 0-9, _` and on Win it's case insensitive, so rather use base32 encoding for generic strings instead of base64)

**Reply by oldtimehacker** [reply]
I keep seeing posts like this, and no answers. How can next.js "go against" basic global persistence and global variable semantics.
Is it a bug or a feature?  Why is it like this? I've found problems with this same topic and I've never seen an answer to this question.
I'm shocked it hasn't been answered, and it leaves me wondering whether this project is commercial ready in any way.
